src/dftbp/dftb/sparse2dense.F90

```diff
@@ -3227,4 +3227,41 @@ subroutine getSparseDescriptor(iNeighbour, nNeighbourSK, img2CentCell, orb, iPai
 
   end subroutine getSparseDescriptor
 
+
+  !> Calculates number of elements in sparse arrays like the real space overlap.
+  subroutine getSparseSize(iNeighbour, nNeighbourSK, img2CentCell, orb, sparseSize)
+
+    !> Neighbours of each atom
+    integer, intent(in) :: iNeighbour(0:,:)
+
+    !> Number of neighbours of each atom
+    integer, intent(in) :: nNeighbourSK(:)
+
+    !> Indexing for mapping image atoms to central cell
+    integer, intent(in) :: img2CentCell(:)
+
+    !> Atomic orbital information
+    type(TOrbitals), intent(in) :: orb
+
+    !> Total number of elements in a sparse structure (ignoring extra indices like spin)
+    integer, intent(out) :: sparseSize
+
+    integer :: nAtom, mNeighbour
+    integer :: ind, iAt1, nOrb1, iNeigh1, nOrb2
+
+    nAtom = size(iNeighbour, dim=2)
+    mNeighbour = size(iNeighbour, dim=1)
+
+    ind = 0
+    do iAt1 = 1, nAtom
```

Comment:
Worth an OMP loop?

Reply:
See https://github.com/dftbplus/dftbplus/pull/1389#discussion_r1487482139_ and https://github.com/dftbplus/dftbplus/pull/1389#discussion_r1487516896_.